src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/GraphNodes/DependenciesGraphProvider.cs

```diff
@@ -105,7 +109,10 @@ public void BeginGetGraphData(IGraphContext context)
                 finally
                 {
                     // OnCompleted must be called to display changes 
-                    context.OnCompleted();
+                    if (anyChanges)
+                    {
+                        context.OnCompleted();
```

Comment:
What's the downside to calling `OnCompleted` if there are no changes?

Reply:
This seems good, but agree that it would be nice to know that the assumption that `OnCompleted` is expensive has been validated.